contrib/platform/src/com/sun/jna/platform/win32/Cfgmgr32Util.java

```diff
@@ -0,0 +1,92 @@
+/* Copyright (c) 2018 Daniel Widdis, All Rights Reserved
+ *
+ * The contents of this file is dual-licensed under 2 
+ * alternative Open Source/Free licenses: LGPL 2.1 or later and 
+ * Apache License 2.0. (starting with JNA version 4.0.0).
+ * 
+ * You can freely decide which license you want to apply to 
+ * the project.
+ * 
+ * You may obtain a copy of the LGPL License at:
+ * 
+ * http://www.gnu.org/licenses/licenses.html
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "LGPL2.1".
+ * 
+ * You may obtain a copy of the Apache License at:
+ * 
+ * http://www.apache.org/licenses/
+ * 
+ * A copy is also included in the downloadable source code package
+ * containing JNA, in file "AL2.0".
+ */
+package com.sun.jna.platform.win32;
+
+import com.sun.jna.Memory;
+import com.sun.jna.Native;
+import com.sun.jna.ptr.IntByReference;
+
+/**
+ * Cfgmgr32 utility API.
+ * 
+ * @author widdis[at]gmail[dot]com
+ */
+public abstract class Cfgmgr32Util {
+    @SuppressWarnings("serial")
+    public static class Cfgmgr32Exception extends RuntimeException {
+        private final int errorCode;
+
+        public Cfgmgr32Exception(int errorCode) {
+            this.errorCode = errorCode;
+        }
+
+        public int getErrorCode() {
+            return errorCode;
+        }
+    }
+
+    /**
+     * Utility method to call Cfgmgr32's CM_Get_Device_ID that allocates the
+     * required memory for the Buffer parameter based on the type mapping used,
+     * calls to CM_Get_Device_ID, and returns the received string.
+     * 
+     * @param devInst
+     *            Caller-supplied device instance handle that is bound to the
+     *            local machine.
+     * @return The device instance ID string.
+     */
+    public static String CM_Get_Device_ID(int devInst) {
```

Comment:
Please add `throws Cfgmgr32Exception` to the signature. It does not need to be there, but I think it is clearer, that the caller should be prepared for the exception.